gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/CopyJobInfoTest.java

```diff
@@ -0,0 +1,139 @@
+
+  @Test
+  public void testToBuilder() {
+    compareCopyJobInfo(COPY_JOB, COPY_JOB.toBuilder().build());
+    compareCopyJobInfo(COPY_JOB_INFO_MULTIPLE_TABLES,
+        COPY_JOB_INFO_MULTIPLE_TABLES.toBuilder().build());
+    CopyJobInfo job = COPY_JOB.toBuilder()
+        .destinationTable(TableId.of("dataset", "newTable"))
+        .build();
+    assertEquals("newTable", job.destinationTable().table());
+    job = job.toBuilder().destinationTable(DESTINATION_TABLE).build();
+    compareCopyJobInfo(COPY_JOB, job);
+  }
+
+  @Test
+  public void testToBuilderIncomplete() {
+    CopyJobInfo job = CopyJobInfo.of(DESTINATION_TABLE, SOURCE_TABLES);
+    compareCopyJobInfo(job, job.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(JOB_ID, COPY_JOB_INFO_MULTIPLE_TABLES.jobId());
+    assertEquals(JOB_STATUS, COPY_JOB_INFO_MULTIPLE_TABLES.status());
+    assertEquals(DESTINATION_TABLE, COPY_JOB_INFO_MULTIPLE_TABLES.destinationTable());
+    assertEquals(SOURCE_TABLES, COPY_JOB_INFO_MULTIPLE_TABLES.sourceTables());
+    assertEquals(CREATE_DISPOSITION, COPY_JOB_INFO_MULTIPLE_TABLES.createDisposition());
+    assertEquals(WRITE_DISPOSITION, COPY_JOB_INFO_MULTIPLE_TABLES.writeDisposition());
+    assertEquals(JOB_ID, COPY_JOB.jobId());
+    assertEquals(JOB_STATUS, COPY_JOB.status());
+    assertEquals(DESTINATION_TABLE, COPY_JOB.destinationTable());
+    assertEquals(ImmutableList.of(SOURCE_TABLE), COPY_JOB.sourceTables());
+    assertEquals(CREATE_DISPOSITION, COPY_JOB.createDisposition());
+    assertEquals(WRITE_DISPOSITION, COPY_JOB.writeDisposition());
+    assertTrue(COPY_JOB.dryRun());
+    assertEquals(JOB_STATISTICS, COPY_JOB.statistics());
+  }
+
+  @Test
+  public void testToPbAndFromPb() {
+    assertNotNull(COPY_JOB.toPb().getConfiguration().getCopy());
+    assertNull(COPY_JOB.toPb().getConfiguration().getExtract());
+    assertNull(COPY_JOB.toPb().getConfiguration().getLoad());
+    assertNull(COPY_JOB.toPb().getConfiguration().getQuery());
+    assertNull(COPY_JOB.toPb().getConfiguration().getCopy().getSourceTables());
+    assertEquals(JOB_STATISTICS, JobStatistics.fromPb(COPY_JOB.statistics().toPb()));
+    assertNull(COPY_JOB_INFO_MULTIPLE_TABLES.toPb().getConfiguration().getCopy().getSourceTable());
+    compareCopyJobInfo(COPY_JOB, CopyJobInfo.fromPb(COPY_JOB.toPb()));
+    compareCopyJobInfo(COPY_JOB, (CopyJobInfo) JobInfo.fromPb(COPY_JOB.toPb()));
+    compareCopyJobInfo(COPY_JOB_INFO_MULTIPLE_TABLES,
+        CopyJobInfo.fromPb(COPY_JOB_INFO_MULTIPLE_TABLES.toPb()));
+    compareCopyJobInfo(COPY_JOB_INFO_MULTIPLE_TABLES,
+        (CopyJobInfo) JobInfo.fromPb(COPY_JOB_INFO_MULTIPLE_TABLES.toPb()));
+    CopyJobInfo job = CopyJobInfo.of(DESTINATION_TABLE, SOURCE_TABLES);
+    compareCopyJobInfo(job, CopyJobInfo.fromPb(job.toPb()));
+    compareCopyJobInfo(job, (CopyJobInfo) JobInfo.fromPb(job.toPb()));
+  }
+
+  private void compareCopyJobInfo(CopyJobInfo expected, CopyJobInfo value) {
```

Comment:
In `BigQueryError` (line 38), you test `hashCode` by including it in `compareBigQueryError`.  I think that's a good idea; we could add a `hashCode` comparison here and in all the other test classes' `compare` functions.